docs/faq.md

```diff
@@ -0,0 +1,47 @@
+# Frequently Asked Questions
+
+This page contains answers to common questions and issues with using Fx.
+
+## Does the order of `fx.Option`s matter?
+
+No, the order in which you provide various Fx options
+to `fx.Options`, `fx.New`, `fx.Module`, and others does not matter.
```

Comment:
I don't think this is true for determining the order in which `fx.Invoke`d functions get run, which was a recent point of confusion (ref: https://github.com/uber-go/fx/issues/1152). Consider exempting Invoke from this statement.

Reply:
Fair. There's a weird situation here.
On paper, there's no guarantee about the order in which invokes run.
But you're right, their order matters.

Reply:
@JacobOaks I realized that this was actually too simplified and lost some information.
Ordering of options relative to fx.New is irrelevant, but ordering of operations relative to each other can matter.
I've rewritten this section to better clarify this.

LMK what you think

Reply:
Actually, looks like we do document the fx.Invoke invocation order.
I'll update this entry.